packages/core/src/scuttle.js

```diff
@@ -0,0 +1,186 @@
+/**
+ * @typedef {object} ScuttleOpts
+ * @property {boolean} enabled - Whether scuttling is enabled or not.
+ * @property {Array<string|RegExp>} exceptions - List of properties to exclude from scuttling.
+ * @property {string} scuttlerName - Name of the scuttler function to use which is expected to be found as a
+ * property on the global object (e.g. if scuttlerName is 'x', scuttler function is obtained from globalThis['x']).
+ */
+
+/**
+ * @typedef {object} GlobalRef
+ * @property {Record<string, any>} [globalThis] - Reference to the global object.
+ */
+
+const { Object, Array, Error, RegExp, Set, console, Proxy, Reflect } =
+  globalThis
+
+const {
+  assign,
+  getOwnPropertyNames,
+  getOwnPropertyDescriptor,
+  create,
+  defineProperty,
+} = Object
+
+const { isArray, from } = Array
+
+const { getPrototypeOf } = Reflect
+
+const { warn } = console
+
+function generateInvokers(prop) {
+  return { get, set }
+  function set() {
+    warn(
+      `LavaMoat - property "${prop}" of globalThis cannot be set under scuttling mode. ` +
+        'To learn more visit https://github.com/LavaMoat/LavaMoat/pull/360.'
+    )
+  }
+  function get() {
+    throw new Error(
+      `LavaMoat - property "${prop}" of globalThis is inaccessible under scuttling mode. ` +
+        'To learn more visit https://github.com/LavaMoat/LavaMoat/pull/360.'
+    )
+  }
+}
+
+/**
+ * Applies scuttling, with the default set of options, including using Snow if passed in as scuttlerFunc.
+ * Scuttle globalThis right after we used it to create the root package compartment.
+ *
+ * @param {GlobalRef} globalRef - Reference to the global object.
+ * @param {ScuttleOpts} opts - Scuttling options.
+ */
+function scuttle(globalRef, opts) {
+  const scuttleOpts = generateScuttleOpts(globalRef, opts)
+
+  if (scuttleOpts.enabled) {
+    if (!isArray(scuttleOpts.exceptions)) {
+      throw new Error(
+        `LavaMoat - exceptions must be an array, got ${typeof scuttleOpts.exceptions}`
+      )
+    }
+    scuttleOpts.scuttlerFunc(globalRef, (realm) =>
+      performScuttleGlobalThis(realm, scuttleOpts.exceptions)
+    )
+  }
+}
+
+/**
+ * @param {GlobalRef} globalRef - Reference to the global object.
+ * @param {ScuttleOpts|boolean} originalOpts - Scuttling options. Accepts `true` for backwards compatibility.
+ * @returns {ScuttleOpts} - Final scuttling options.
+ */
```

Comment:
```suggestion
/**
 * @overload
 * @param {GlobalRef} globalRef - Reference to the global object.
 * @param {ScuttleOpts} originalOpts - Scuttling options
 * @returns {ScuttleOpts} - Final scuttling options.
 */
 
 /**
 * @overload
 * @deprecated
 * @param {GlobalRef} globalRef - Reference to the global object.
 * @param {boolean} originalOpts - Deprecated; use options instead
 * @returns {ScuttleOpts} - Final scuttling options.
 */
 
 /**
 * @param {GlobalRef} globalRef - Reference to the global object.
 * @param {ScuttleOpts|boolean} originalOpts - Scuttling options
 * @returns {ScuttleOpts} - Final scuttling options.
 */
```

To @legobeat's comment, try this. I'm not sure if it'll mark the entire function deprecated, or just that one usage of it.  But if it's the latter then 👍 .  add a `// @ts-check` at the top of the file to see what it thinks (you can remove later if you want)

Reply:
🥲 
![Screenshot 2024-01-31 at 12 28 31](https://github.com/LavaMoat/LavaMoat/assets/13243797/150ca4bc-3d16-4ca9-bd7f-fed2adc61593)

Reply:
I wasn't able to learn online how to achieve this.. thoughts? @boneskull

Reply:
Decided with @boneskull that this might not be easy to achieve, we can circle back to this in a different PR